### DISCUSSION TITLE
[Question]: beginner question: how does the frigate UI render video clips, in contrast to live camera streams?

### DISCUSSION THREAD

**Discussion by ElectronicBattle** [blakeblackshear/frigate#12704] [Ask A Question] [answered]
### What is your question:

Hello all

I have become very confused. 

My Frigate _is_ working but I don't understand why my browser, running on my  main PC, is unable to render the _video clips_ in Frigate's UI served out by TrueNAS box. 

My main PC can render Frigate's _live_ camera streams perfectly well e.g.  via MotionEye which handles all eight cameras (a subset of four of them simultaneously feed frigate). It can easily playback the saved recordings using vlc or ffplay or mplayer etc.  

Similarly, I have already tested my main PC's web various browsers (Chrome/Chromium/Edge/Firefox/Opera) against a variety of hvec test pages with embedded hvec videos and they work.

This is my setup:

![image](https://github.com/user-attachments/assets/cef5788f-6bb7-4a10-b22b-9ccf02555e32)

- The "basic" Nvidia GPU in the TrueNAS can do ndec decoding but not nvenc encoding and is hence unable to transcode
- TrueNAS is hosting frigate **0.13** via some form of built-in Docker "app" and cannot be changed to 0.14 at the moment
- The cameras are fixed as H265 (hvec) - they can't be switched to h264
- My  Radeon RX580: https://www.techpowerup.com/gpu-specs/radeon-rx-580.c2938

I can't understand why I would need to task what is too "basic" a GPU in TrueNAS to transcode from h265(hvec) to h264 in order for my main PC browsers to be able to render the video clips.

1. What am I misunderstanding?
2. how exactly *does* the video clip playback rendering mechanism work in the UI?
3. how does 2. differ from the live camera stream rendering?
4. Is there a way to take the video clip playback resource/URL/thing and pipe it instead through (e.g.) mplayer or vlc etc.?

**Comment by hawkeye217** [comment, marked as answer]
Live video and recordings playback use two different technologies (MSE, WebRTC, or jsmpeg for live and HLS for recordings playback) and thus two different player types in frigate.

**Reply by ElectronicBattle** [reply]
I can render the four live streams via mse or webRTC or jsmpeg so that is working properly and your answer certainly helps with questions 2 & 3 and gives me something to work with and try to learn - thanks. 

Q2 and Q3 **Conclusion:I need to get HLS working for clip playback.**

It seems as though I ought to be able to do rendering/transcoding on my main PC without needing the TrueNAS box to do much (other than run the docker image with the Coral h/w).

If I can eventually find out the answers to my  remaining questions 1 & 4 it would certainly help direct my efforts and I might be able to solve this problem (or at least lay it to rest!).     

Thanks

EB

**Reply by hawkeye217** [reply]
> It seems as though I ought to be able to do rendering/transcoding on my main PC without needing the TrueNAS box to do much (other than run the docker image with the Coral h/w).

You could set up go2rtc on another, more capable machine to do the transcoding, and then point your frigate inputs at it.

**Reply by ElectronicBattle** [reply]
In the frigate webUI, the built-in video player which activates when you select events|past events and then select a clip ...that must be pointing at, or is being fed, a resource for it to consume. Is that resource exposed somewhere so I can point (e.g.) vlc at it instead, or a web browser player of some sort?

**Reply by NickM-27** [reply]
You can point VLC at the m3u8 playlist. Just get the url from the browser network tab

**Reply by ElectronicBattle** [reply]
@hawkeye217 - I like the idea of a separate instance of go2rtc and I have started looking into that.
@NickM-27 - I have found the m3u8 and it works (vlc, mplayer etc.). which also gives me a sense of direction now.

And finally, referring to my diagram above, please can either of you confirm that if I want to watch the clips on my "main PC"

a - there is no need for me to transcode on my TrueNAS box

b - there is no  need for me to transcode on my main PC 

c - it should be possible for me to come up with  a mechanism where the existing clips are played on demand from the UI but in (e.g.) vlc or ffplay etc. rather than being played in the UI window as per the current implementation?
